NewForce-at-Mountwest/bangazon-workforce-management-the-final-slappers
# Added Index, Details, and Create views for Departments resource

# Description

I created a `DepartmentsController.cs` file to allow the HR to view all departments, create a department, and view a single department's details with a list of employees.
+ An `Index,cshtml` file was added to view a list of all departments
+ A `Details.cshtml` file was added to view a single department's details
+ A `Create.cshtml` file was added to create a new department with a form with fields of 'Name' and 'Budget'

Before starting on `DepartmentsController.cs`, I seeded my `BangazonWorkforce` SQL database with appropriate data.

## Related Ticket(s)
https://trello.com/b/vc9MSx7t/the-final-slappers
+ Ticket #5- HR should be able to view all departments
+ Ticket #6- HR should be able to create a department
+ Ticket #7- HR should be able to view a department's details
+ Ticket #12- Department details should itemize teammates

## Steps to Test
1. After making sure you have our repo cloned down into a local repository, `git fetch --all` in the `master` branch to retrieve this branch named `sm-departments` that I have pushed up.

2. `git checkout sm-departments` to switch to my branch and open up Visual Studio

3. Seed your database with the data posted in this SQL Query.

+ Follow this link to see the SQLQuery file with the data: 
    https://github.com/NewForce-at-Mountwest/bangazon-workforce-management-the-final-slappers/blob/sm-departments/SQLQuery1.sql

4. Then, run the application in a browser window. You should see the 'Home' page with a navigation bar at the top of the page. Select '_Departments_' in the navigation bar. This should redirect you to the _Departments_ Index view. You should see a table with 3 departments, each with a name, budget, a column for department size of number of employees in the department, and a _Details_ badge/button for each department.

5. Select the _Details_ button for a department. You should be redirected to the single department's details page where you should see the department name as the page header, and below that, the department name, department budget, and a list of all the employees working in the department with their first and last names should be displayed. Also, underneath, each department details, there should be a _Back To List_ link that redirects the user back to the Index view if selected.

6. Select the _Create New_ link on the Departments Index view. You should be redirected to a _Create New Department_ form with the form fields of _Name_ and _Budget_ to add a department name and department budget. There should be a _Create_ button to create the new department and redirect the user back to Index view to see their new department in the table, and a _Back To List_ link that redirects the user back to the Index view if they would no longer like to create a new department.


# Checklist:
- [x] I have opened up `sm-departments` in my local repository.
- [x] I have seeded my SQL database with the data provided above, if applicable.
- [x] After running the application, clicking on _Departments_ in the nav bar redirects me to the Departments' Index view.
- [x] The Index view displays a table with 3 departments, each with a name, budget, and department size.
- [x] Select _Details_ for a department. The Details view should have department name as the header, and display the department name, the department budget, and the list of employees working in the department with their first and last name.
- [x] Select _Create New_ in the list view. The Create form should offer form fields to add a department name and a department budget.
- [x] I have celebrated this code's functionality with a "Hip Hip Hooray" and if available, an enthusiastic throw of confetti!
